teps/0007-conditions-beta.md

```diff
@@ -0,0 +1,612 @@
+
+## Requirements
+<!--
+List the requirements for this TEP.
+-->
+1. Users need a way to specify complex `Conditions`, such as those needing access to files in a `Workspace`. 
+1. Users need a way to specify simple `Conditions`, such as checking the name of a git branch matches.
+1. Users need a way to specify what happens when `Conditions` evaluate to `False`.
+
+## Proposal
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+For _simplicity_, we propose deprecating the separate `Conditions` CRD and use `Tasks` instead for guarded execution of other `Tasks`. For _efficiency_, we propose using string expressions through Kubernetes `MatchExpressions` and Tekton’s `CustomTasks` to perform simple checks for guarded execution of `Tasks` without spinning up new `Pods`. For _skipping_, we propose adding a boolean to allow users to specify whether to skip the guarded `Task` only or to skip the guarded `Task` and its dependent `Tasks`. 
+
+This design will give us the flexibility to handle both simple `Guards` (e.g. checking a branch name) using string expressions and complex `Guards` (e.g. those that need a `Workspace` and/or `Volume` with the files to execute on) in a `Pod`, while providing more options when a `Guard` evaluates to `False`. 
+
+
+### Simplicity
+
+As discussed in the background section, `Conditions` manifest themselves as `Tasks`. We want to keep Tekton as simple as possible by reusing existing components. So we propose phasing out the separate `Conditions` CRD and eventually deprecating it. In place of `Conditions`, we want to use `Tasks` - referred to as `Guards` - to determine whether or not a `Task` executes. We refer to them as `Guards` because they determine if a `Task` executes, not if/else as would be expected from a `Condition`; more details on `Guards` vs `Conditions` naming can be found in [this issue](https://github.com/tektoncd/pipeline/issues/2635). Thus, we won’t have `Conditions` to migrate to beta and won’t have to maintain the separate `Conditions` CRD. Moreover, the features supported by `Tasks` become readily available to be used for guarded execution of `Tasks`. 
+
+A `Task` that is used as a `Guard` will produce a `Result` called `Skip` with values `True` or `False` -- to indicate whether the `Task` it’s guarding will be executed or not. When we have more than one `Guard`, the `Task` will be executed when all the `Guards` evaluate to `True`. Using `Results` allows us to distinguish between when a `Task` fails and when the `Guard` evaluates to `False`. 
```

Comment:
it might be worth specify exactly the string values expected here: afaik both yaml and json actually do have bools, but we're talking about string values here

Reply:
question: what happens if a user provides another value? im thinking we'd treat that as an error and fail the guard + pipeline